addons/payment_redsys/models/payment_transaction.py

```diff
@@ -0,0 +1,221 @@
+
+        Note: self.ensure_one() from `_get_processing_values`.
+
+        :param dict processing_values: The generic processing values of the transaction.
+        :return: The dict of provider-specific processing values.
+        :rtype: dict
+        """
+        res = super()._get_specific_rendering_values(processing_values)
+        if self.provider_code != 'redsys':
+            return res
+
+        merchant_parameters = self._prepare_merchant_parameters()
+        signature = self._redsys_calculate_signature(
+            merchant_parameters,
+            self.reference,
+            self.provider_id.redsys_secret_key,
+        )
+        return {
+            'api_url': self.provider_id._redsys_get_api_url(),
+            'signature_version': self.provider_id.signature_version,
+            'merchant_parameters': merchant_parameters,
+            'signature': signature,
+        }
+
+    def _prepare_merchant_parameters(self):
+        """ Create the merchant parameters for the request to the Redsys API.
+
+        :return: The encoded merchant parameters.
+        :rtype: str
+        """
+        # All parameters are of type string, the amount is expressed in cents
+        amount = str(int(self.amount * 100))
+        base_url = self.provider_id.get_base_url()
+        webhook_url = urls.url_join(base_url, RedsysController._webhook_url)
+        return_url = urls.url_join(base_url, RedsysController._return_url)
+        payment_method = const.PAYMENT_METHODS_MAPPING.get(self.payment_method_id.code, 'C')
+
+        merchant_parameters = {
+            "DS_MERCHANT_AMOUNT": amount,
+            "DS_MERCHANT_CURRENCY": CURRENCY_4217_MAPPING[self.currency_id.name],
+            "DS_MERCHANT_MERCHANTCODE": self.provider_id.redsys_merchant_code,
+            "DS_MERCHANT_MERCHANTURL": webhook_url,
+            "DS_MERCHANT_ORDER": self.reference,
+            "DS_MERCHANT_TERMINAL": self.provider_id.redsys_merchant_terminal,
+            "DS_MERCHANT_TRANSACTIONTYPE": "0",  # Authorization
+            "DS_MERCHANT_URLKO": return_url,
+            "DS_MERCHANT_URLOK": return_url,
+            "DS_MERCHANT_PAYMETHODS": payment_method,
+            "DS_MERCHANT_EMV3DS": {
+                "billAddrCity": self.partner_city,
+                "billAddrCountry": COUNTRY_3166_MAPPING.get(self.partner_country_id.code, ''),
+                "billAddrLine1": self.partner_address,
+                "billAddrPostCode": self.partner_zip,
+                "billAddrState": self.partner_state_id.code,
+                "cardholderName": self.partner_name,
+                "email": self.partner_email,
+            }
+        }
+
+        # Convert parameters to JSON and encode in BASE64
+        encoded_merchant_parameters = base64.b64encode(
+            json.dumps(merchant_parameters).encode()
+        ).decode()
+
+        return encoded_merchant_parameters
+
+    def _compute_reference(self, provider_code, prefix=None, separator='-', **kwargs):
```

Comment:
Okay, don't know how I missed that, but this should be right at the beginning of this class, as it is the first to be called.   And it should have a proper decorator.
```suggestion
    @api.model
    def _compute_reference(self, provider_code, prefix=None, separator='-', **kwargs):
```